plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -316,42 +334,55 @@ def _get_agent(self, agent_input: str, agent_type: str) -> dict:
 
         :return: dict
         """
+        start_time = datetime.now()
         agents = []
-        payload = {
-            "query": "query( $orgId:String! ) { organization(id: $orgId) { assets( first: 10000 ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } publicIpAddress location { city region countryName countryCode continent } agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
-            "variables": {"orgId": self.org_key},
-        }
-
-        self.logger.info("Getting all agents...")
-        results_object = self._post_payload(payload)
-
-        has_next_page = (
-            results_object.get("data", {})
-            .get("organization", {})
-            .get("assets", {})
-            .get("pageInfo", {})
-            .get("hasNextPage")
-        )
-        agents.extend(self._get_agents_from_result_object(results_object))
-        self.logger.info("Initial agents received.")
-
-        agent = self._find_agent_in_agents(agents, agent_input, agent_type)
-        if agent:
-            return agent
-
-        # See if we have more pages of data, if so get next page and append until we reach the end
-        self.logger.info(f"Extra pages of agents: {has_next_page}")
-        while has_next_page:
+        has_next_page = True
+        if not next_cursor:
+            payload = {
+                "query": "query( $orgId:String! ) { organization(id: $orgId) { assets( first: 10000 ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } publicIpAddress location { city region countryName countryCode continent } agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
+                "variables": {"orgId": self.org_key},
+            }
```

Comment:
is there another issue with how these queries are being used? `get_agent` seems to pass a query ending in 
```
publicIpAddress location { city region countryName countryCode continent } agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }
```
and the same for the next_pages call also, but the one for get_agents_by_ip is using:
```
publicIpAddress agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }
```

Reply:
I think we updated our query to include location details but possibly not in every instance of the query. I'll update this as well.